juicer/workflow/workflow.py

```diff
@@ -1,27 +1,263 @@
+
+        """ Create the tasks Graph and perform topological sorting
+
+            A topological sort is a nonunique permutation of the nodes
+            such that an edge from u to v implies that u appears before
+            v in the topological sort order.
+
+            :return: Return a list of nodes in topological sort order.
+        """
+        # First, map the tasks IDs to their original position
+        tasks_position = {}
+
+        for count_position, task in enumerate(self.workflow_data['tasks']):
+            tasks_position[task['id']] = count_position
+
+        sorted_tasks_id = nx.topological_sort(self.workflow_graph, reverse=False)
+
+        return sorted_tasks_id
+
+    def check_null_source_id_tasks(self):
+        for flow in self.workflow_data['flows']:
+            if flow['source_id'] == "":
+                return 0
+        return 1
+
+    def check_null_target_id_tasks(self):
+        for flow in self.workflow_data['flows']:
+            if flow['target_id'] == "":
+                return 0
+        return 1
+
+
+    def get_reversed_graph(self):
+        """
+            Return the reverse of the graph.
+            The reverse is a graph with the same nodes
+            and edges but with the directions of the edges reversed.
+        """
+        workflow_graph_reversed = self.workflow_graph.reverse()
+
+        return workflow_graph_reversed
+
+
+    def get_all_ports_operations_tasks(self):
+        return 0
+
+    def get_ports_from_operation_tasks(self, id_operation):
+        # Can i put this information here?
+        params = {
+            'base_url':'http://beta.ctweb.inweb.org.br',
+            'item_path': 'tahiti/operations',
+            'token': '123456',
+            'item_id': id_operation
+        }
+
+        # Querying tahiti operations to get number of inputs and outputs
+        x = tahiti_service.query_tahiti_operations(params['base_url'],
+                                                   params['item_path'],
+                                                   params['token'],
+                                                   params['item_id'])
+        # Get operation requirements in tahiti
+        result = {
+                    'N_INPUT': 0,
+                    'N_OUTPUT': 0,
+                    'M_INPUT': 'None',
+                    'M_OUTPUT': 'None'
+                 }
+
+        for y in x['ports']:
+            if y['type'] == 'INPUT':
+                result['M_INPUT'] = y['multiplicity']
+                if 'N_INPUT' in result:
+                    result['N_INPUT'] += 1
+                else:
+                    result['N_INPUT'] = 1
+            elif y['type'] == 'OUTPUT':
+                result['M_OUTPUT'] = y['multiplicity']
+                if 'N_OUTPUT' in result:
+                    result['N_OUTPUT'] += 1
+                else:
+                    result['N_OUTPUT'] = 1
+        return result
+
+    # only to debug
+    def verify_outdegree_edges(self, atr):
```

Comment:
Be consistent with names: check, verify. Maybe instead of using check earlier in the code, call it as is_there_null_source_id, because it is not really a check.